src/main/java/ch/njol/skript/lang/util/ConvertedExpression.java

```diff
@@ -265,11 +264,8 @@ public Expression<?> getSource() {
 	}
 
 	@Override
-	@SuppressWarnings("unchecked")
 	public Expression<? extends T> simplify() {
-		Expression<? extends T> convertedExpression = source.simplify().getConvertedExpression(to);
-		if (convertedExpression != null)
-			return convertedExpression;
+		source = source.simplify();
```

Comment:
Why this change? Calling getConvertedExpression again could potentially result in a Literal of the desired (conversion) type, which seems like something we'd want. Or is the concern over losing the source information?